### PR TITLE
Upgrade deps and fix netlify

### PR DESCRIPTION
**Pull request by poteirard**
- Fix Netlify to support PNPM
- Upgrade react-scripts and react-testing-library
- Remove jest-environment-jsdom-sixteen

### CODE REVIEW

File: src/components/atoms/Text/Text.tsx [hunk 1]
```diff
@@ -64,7 +64,7 @@ const Text = styled.span<TextProps>`
 
   ${({ className = '', as }) =>
     (!as || as === 'span') &&
-    className.split(' ').some((clss) => /[mp](\:[mp])?[tblrxy]?-\d+/.test(clss)) &&
+    className.split(' ').some((clss) => /[mp](:[mp])?[tblrxy]?-\d+/.test(clss)) &&
```
Comment (steedems) [inline src/components/atoms/Text/Text.tsx:67]: Was this change intended?

File: netlify.toml [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# Workarround netlify with pnpm
+# See: https://github.com/netlify/build/issues/1633
```
Comment (steedems) [inline netlify.toml:2]: 👀 👍

**Review by steedems — APPROVED** [review]

File: src/components/atoms/Text/Text.tsx (hunk 1 above)
Comment (poteirard) [reply src/components/atoms/Text/Text.tsx:67]: Yep, new eslint check
